solarforecastarbiter/io/reference_observations/srml.py

```diff
@@ -0,0 +1,235 @@
+    1002. These columns represent GHI at instrument 1 and
+    instrument 2 respectively. The `pvlib.iotools` package
+    converts these to 'ghi_1' and 'ghi_2' for us. We use these
+    labels to differentiate between measurements recorded by
+    different instruments.
+    """
+    start = pd.Timestamp.now()
+    end = pd.Timestamp.now()
+    extra_params = common.decode_extra_parameters(site)
+    # use site name without network here to build
+    # a name with the original column label rather than
+    # the SFA variable
+    site_name = common.site_name_no_network(site)
+    try:
+        site_df = fetch(api, site, start, end)
+    except error.HTTPError:
+        logger.error('Could not find data to create observations '
+                     f'for SRML site {site_name}.')
+        return
+    else:
+        if site_df is None:
+            logger.error('Could not find data to create observations '
+                         f'for SRML site {site_name}.')
+            return
+        for variable in srml_variable_map.keys():
+            matches = [col for col in site_df.columns if variable in col]
+            for match in matches:
+                observation_extra_parameters = extra_params.copy()
+                observation_extra_parameters.update({
+                    'network_data_label': match})
+                try:
+                    # Here, we pass a name with match instead of variable
+                    # to differentiate between multiple observations of
+                    # the same variable
+                    common.create_observation(
+                        api, site, srml_variable_map[variable],
+                        name=f'{site_name} {match}',
+                        interval_label='beginning',
+                        extra_params=observation_extra_parameters)
+                except HTTPError as e:
+                    logger.error(
+                        f'Failed to create {variable} observation at Site '
+                        f'{site.name}. Error: {e.response.text}')
+
+
+def update_observation_data(api, sites, observations, start, end):
+    """Post new observation data to a list of SRML Observations
+    from start to end.
+
+    api : solarforecastarbiter.io.api.APISession
+        An active Reference user session.
+    sites: list
+        List of all reference sites as Objects
+    start : datetime
+        The beginning of the period to request data for.
+    end : datetime
+        The end of the period to request data for.
+    """
+    srml_sites = common.filter_by_networks(sites, 'UO SRML')
+    for site in srml_sites:
+        obs_df = fetch(api, site, start, end)
+        site_observations = [obs for obs in observations if obs.site == site]
+        for obs in site_observations:
+            obs_params = common.decode_extra_parameters(obs)
+            var_label = obs_params['network_data_label']
+            var_df = obs_df[[var_label]]
```

Comment:
move to common, possible adding a start, end parameter

Reply:
Do you mean just this DataFrame creation code here, or including the handling for forward filled data below?

Reply:
I mean all of it. It will take a bit more than I orginally thought, but still probably good to avoid the duplication. You'll need to probably add start, end, variable_name (default observation.variable) to ``common.post_observation_data``. Then from here, start=start, end=min(end, pd.timestamp.now). Do you need the last_valid_index check when dropping nans?